src/language/from_relay/mod.rs

```diff
@@ -655,29 +671,51 @@ fn compile_expression(
         .clone()
         .downcast::<tvm::ir::relay::TupleGetItem>()
     {
-        // We've only implemented tuple functionality for handling batch norms,
-        // at the moment. So we check to make sure this is handling a batch
-        // norm.
-
-        let arg = tuple_get_item
+        if let Ok(call) = tuple_get_item
             .tuple
             .clone()
             .downcast::<tvm::ir::relay::Call>()
-            .unwrap();
-        assert_eq!(
-            arg.op
-                .clone()
-                .upcast::<tvm::ir::expr::BaseExpr>()
-                .downcast::<tvm::ir::op::Op>()
-                .unwrap()
-                .name
-                .as_str(),
-            Ok("nn.batch_norm")
-        );
-        assert_eq!(tuple_get_item.index, 0);
+        {
+            // For batch norms, we use a hack where if
+            // simplify_batch_norm_for_inference_hack is set,
+            // we compile the Relay batch norm to a single output.
+
+            // All other expressions are compiled using Language::TupleGetItem construct
+
+            if simplify_batch_norm_for_inference_hack
+                && call
+                    .op
+                    .clone()
+                    .upcast::<tvm::ir::expr::BaseExpr>()
+                    .downcast::<tvm::ir::op::Op>()
+                    .unwrap()
+                    .name
+                    .as_str()
+                    .unwrap()
+                    == "nn.batch_norm"
+                && tuple_get_item.index == 0
+            {
+                get_compiled_expression(tuple_get_item.tuple.clone())
+            } else {
+                let data_id = get_compiled_expression(tuple_get_item.tuple.clone());
+                let index_id = glenside_expr.add(Language::Usize(tuple_get_item.index as usize));
+                glenside_expr.add(Language::TupleGetItem([data_id, index_id]))
+            }
+        } else {
```

Comment:
Can you add a simple comment here, saying that this is the standard case, where Relay `TupleGetItem`s are converted directly to Glenside `TupleGetItem`s? I just want to make it clear that the `if` branch is a corner case.